packages/sanes-browser-extension/src/extension/background/model/persona/index.ts

```diff
@@ -200,17 +222,55 @@ export class Persona {
     );
     this.jsonRpcSigningServer = new JsonRpcSigningServer(this.core);
 
-    this.subscribeToTxUpdates(transactionsUpdater);
+    this.subscribeToBalancesUpdates(balancesUpdater);
   }
 
-  private subscribeToTxUpdates(notifyApp: (transactions: readonly ProcessedTx[]) => void): void {
-    this.core.signedAndPosted.updates.subscribe({
-      next: async signedAndPosted => {
-        const processed = await Promise.all(signedAndPosted.map(s => this.processTransaction(s)));
-        const filtered = processed.filter(isNonNull);
-        notifyApp(filtered);
-      },
-    });
+  private getActiveConnections(): readonly BlockchainConnection[] {
+    return [...this.connections.values()];
+  }
+
+  private async subscribeToBalancesUpdates(
+    notifyApp: (balances: readonly (readonly Amount[])[]) => void,
+  ): Promise<void> {
+    const config = await getConfigurationFile();
+    const chains = config.chains;
+    const connections = this.getActiveConnections();
+
+    const accounts = await this.accountManager.accounts();
+    const identities = accounts.flatMap(account => account.identities);
+
+    for (const connection of connections) {
+      const chain = chains.find(chain => chain.chainSpec.chainId === connection.chainId());
+      if (!chain) {
+        throw new Error(`Chain for ${connection.chainId()} not found.`);
+      }
+      const identity = identities.find(identity => identity.chainId === connection.chainId());
+      if (!identity) {
+        continue;
+      }
+
+      // TODO: Please invesitage this issue. If you will remove this you will get:
+      // "Error: This RPC client type cannot subscribe to events".
+      // Please consult with @webmaster128
+      if (chain.chainSpec.chainId === "local-iov-devnet") {
+        continue;
+      }
```

Comment:
@abefernan please investigate this. @webmaster128 can you help with this?

Reply:
Ah, yeah, this is an issue. We do not connect to websockets in the extension because the extension process runs for a very long time. As soon as the websocket connection breaks, the subscriptions breaks.

This is basically blocked by https://github.com/iov-one/iov-core/issues/1200

Reply:
But can we call faucet from the extension? Also is it worth to implement this feature (I mean to showing balance in the extension) at all? @annatorok

Reply:
> But can we call faucet from the extension?

I wonder if we should really do this. Don't think so. It is implemented in the wallet.